src/mainwindow.cpp

```diff
@@ -256,3 +268,19 @@ void MainWindow::aboutHotspot()
     dialog.adjustSize();
     dialog.exec();
 }
+
+void MainWindow::on_systemInfoGroupBox_toggled(bool arg1)
+{
+    hideShowSystemInfo(arg1);
+}
+
+void MainWindow::hideShowSystemInfo(bool visible)
+{
+    foreach (QLabel *label, ui->systemInfoGroupBox->findChildren<QLabel*>()) {
```

Comment:
Looking at the result, I think it would be better to always show the system information, but make the overall summary page scrollable. I.e. remove the toggle switch, always show the contents, but if the window is not tall enough it will not be visible and the user has to scroll to see it.